packages/state-transition/src/block/processRandao.ts

```diff
@@ -28,6 +27,14 @@ export function processRandao(
   }
 
   // mix in RANDAO reveal
-  const randaoMix = xor(Buffer.from(getRandaoMix(state, epoch)), Buffer.from(digest(randaoReveal)));
+  const randaoMix = xor(getRandaoMix(state, epoch), digest(randaoReveal));
   state.randaoMixes.set(epoch % EPOCHS_PER_HISTORICAL_VECTOR, randaoMix);
 }
+
+function xor(a: Uint8Array, b: Uint8Array): Uint8Array {
```

Comment:
https://github.com/ChainSafe/lodestar/pull/6485#issuecomment-1967746399 :D

Reply:
yeah... its finally happening. Also able to use the slightly more efficient [inplace](https://github.com/browserify/buffer-xor/blob/master/inplace.js)